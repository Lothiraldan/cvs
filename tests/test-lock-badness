#!/bin/sh

hg init a
echo a > a/a
hg --cwd a ci -A -m a
hg clone a b
echo b > b/b
hg --cwd b ci -A -m b
chmod 100 a/.hg/store
hg --cwd b push ../a
chmod 700 a/.hg/store
