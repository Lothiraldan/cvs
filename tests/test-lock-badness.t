#if unix-permissions no-root no-windows

Prepare

  $ hg init a
  $ echo a > a/a
  $ hg -R a ci -A -m a
  adding a

  $ hg clone a b
  updating to branch default
  1 files updated, 0 files merged, 0 files removed, 0 files unresolved

One process waiting for another

  $ cat > hooks.py << EOF
  > import time
  > def sleepone(**x): time.sleep(1)
  > def sleephalf(**x): time.sleep(0.5)
  > EOF
  $ echo b > b/b
  $ hg -R b ci -A -m b --config hooks.precommit="python:`pwd`/hooks.py:sleepone" > stdout &
  $ hg -R b up -q --config hooks.pre-update="python:`pwd`/hooks.py:sleephalf"
  waiting for lock on working directory of b held by '*:*' (glob)
  got lock after ? seconds (glob)
  warning: ignoring unknown working parent d2ae7f538514!
  $ wait
  $ cat stdout
  adding b

Pushing to a local read-only repo that can't be locked

  $ chmod 100 a/.hg/store

  $ hg -R b push a
  pushing to a
  searching for changes
  abort: could not lock repository a: Permission denied
  [255]

  $ chmod 700 a/.hg/store
#endif
