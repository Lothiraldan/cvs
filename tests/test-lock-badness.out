adding a
adding b
pushing to ../a
abort: could not lock repository ../a: Permission denied
